### DISCUSSION TITLE
standard-keyboard/

### DISCUSSION THREAD

**Discussion by giscus** [amir-pourmand/hugo-papermod-farsi-template#42] [Announcements]
# standard-keyboard/

چند وقت پیش، همین طوری داشتم تو اینترنت می‌گشتم به این مطلب برخوردم و بعد تو PDFای که صفحه‌کلید استاندارد فارسی رو تصویب کرده بودند، دیدم که علاوه بر Shift کلید Alt (Option در مک) هم برای تایپ کاراکترهای خاص کاربرد داره.
دو تا تصویر بالا کاملاً مشخصه. کارکرد کلید شیفت رو همه می‌دونیم اما کلید Option یا Alt هم کارکردهای مفیدی داره.
مثلاً من قبلاً هنگامی که تو فارسی به کاراکتر # (هشتگ) نیاز داشتم، اول صفحه کلید رو انگلیسی می‌کردم و بعد تایپش می‌کردم.

https://aprd.ir/standard-keyboard/

**Comment by masoudnasrollahi** [comment]
من فقط از کیبورد Enhanced Microsoft Persian استفاده می‌کردم. به جایگیری حروف و کاراکترها عادت کردم. تا اینکه این وسواس سراغم آمد که اعدادم را حتما فارسی تایپ کنم. پس مجبور شدم Persian (Standard) را هم نصب کنم.
جالبه داخل صفحه‌کلید استاندارد کاراکتر مجزایی برای «ۀ» وجود نداره، که البته منطقیه. چون ترکیبیه از «ه» و «ٔ».
قبلا که می‌خواستم از «سکون» استفاده کنم، داخل اینترنت دیده بودم که کلید Alt را بگیرم و شمارۀ ۰۲۵۰ را بگیرم: «ْ».
شما در جریانی این کد را برای کاراکترهای مختلف از کجا میشه پیدا کرد؟ مثلا برای «…» .

**Reply by pourmand1376** [reply]
سلام مسعود جان. وقتت بخیر. 

تو ویندوز دو تا ابزار هست که برای کار کردن با صفحه کلید مفیده:
- Character Map ([+](https://superuser.com/questions/655641/find-alt-code-for-any-character))
- On-Screen Keyboard



اولی با کلید کردن روی کاراکترها همون ALT Code ای که می‌خوای رو بهت میده. البته برای همه حروف نیست. ولی بگردی احتمالاً برای حرف تو هست. ولی جز موارد خاص واقعاً پیشنهاد نمی‌کنم. چون اکثر کاراکترها رو بدون استفاده از اون هم میشه تایپ کرد. حالا یه سری هاش با شیفت فعال میشه و یه سری ها با کلید ALT. مثلاً این کاراکتر سکون با Shift + Q برای من تایپ میشه. 

ولی همون طور که گفتم کاراکترهایی که با alt فعال میشن، تو ویندوز فعلاً به درستی پیاده سازی نشدند. اینه که برای اونها احتمالاً باید همون alt code شون رو پیدا کنی. لینکی که جلوی کاراکتر مپ هم گذاشتم، کمک‌کننده است.

**Reply by masoudnasrollahi** [reply]
سلامت باشی امیر. به نظرم یه پیاده‌روی طولانی یا حتی دوچرخه‌سواری کنارت توی یزد حسابی می‌چسبه!🐾😊

**Reply by pourmand1376** [reply]
حتماً :)) استخر هم دوست دارم. 😁

**Reply by masoudnasrollahi** [reply]
تفاوت و نسبت ALT Code و Unicode چیه؟
***
میشه لطفاً تجربۀ کاربری [صفحۀ جستجوی وبلاگ](https://aprd.ir/search/) را بهتر کنی، باکس وارد کردن کلمات برای جستجو خیلی سخت کلیک و فعال میشه تا بخواد text cursor را نمایش بده.

**Comment by AmooEbrahim** [comment]
آقاجان دمت گرم، کلی وقت بود منم همین کارو می‌کردم و واقعا می‌گفتم آخه چرا دسترسی به هشتگ ندارم. ویندوز انگار اینو ساپورت نمی‌کنه اما برای من که لینوکس استفاده می‌کنم خیلی کارآمد بود.
ممنونم ازت

**Comment by masoudnasrollahi** [comment]
درود. داخل ابسیدین با [افزونۀ Smart Typography](obsidian://show-plugin?id=obsidian-smart-typography) وقتی سه تا نقطه تایپ کنی، خودش تبدیلش می‌کنه به ellipsis …. برای حرفه‌ای‌تر نوشتن، تایپ em dash, en dash را بدون ALT Code، راحت ممکن می‌کنه. اگه [افزونۀ Typographer](obsidian://show-plugin?id=typographer) را نگاه کنی، متوجه کاربرد مهم Auto-pair برای علامت نقل‌قول‌گذاری در زبان خودمون، یعنی گیومه میشی. امیدوارم بهره بردن از و کندوکاو این دو تا افزونه بهت کمک کنه راحت‌تر مرتب و تمیز بنویسی، و ما هم از زیبایی نوشته‌ها لذت ببریم (:
